overlord/snapstate/snapstate.go

```diff
@@ -1482,6 +1490,10 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 
 	// updates is sorted by kind so this will process first core
 	// and bases and then other snaps
+	var transLane int
+	if transactional {
+		transLane = st.NewLane()
```

Comment:
I think it'd be useful to add a managers level test (in overlord/managers_test.go) that executes install/update/remove.

Reply:
Also, it'd be interesting to have a test that runs an update of kernel + base with the flag enabled. AFAIU the code should do the right thing, maybe some tweaks would be required so that tasks (link-snap, auto-connect) do not have duplicated lanes when they get rearranged for single reboot.